providers/sru/bin/dkms_build_validation.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+#
+# Copyright 2017-2024 Canonical Ltd.
+# Written by:
+#   Taihsiang Ho (tai271828) <taihsiang.ho@canonical.com>
+#   Fernando Bravo <fernando.bravo.hernandez@canonical.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from collections import Counter
+import logging
+from packaging import version
+import re
+import subprocess
+import sys
+import textwrap
+from typing import Dict, List
+
+logger = logging.getLogger("dkms_build_validation")
+
+
+def run_command(command: str) -> str:
+    """Run a shell command and return its output"""
+    try:
+        result = subprocess.check_output(
+            command,
+            stderr=subprocess.STDOUT,
+            shell=True,
+            universal_newlines=True,
+        )
+        return result.strip()
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(
+            "Command '{0}' failed with error:\n{1}".format(command, e.output)
+        )
```

Comment:
This is not getting caught anywhere, so it'll cause a backtrace when this, predicted, problem happens. So I'd recommend raising SystemExit with a nice info.